pkg/vsphere/tasks/waiter.go

```diff
@@ -268,6 +268,23 @@ func IsNotFoundError(err error) bool {
 	return false
 }
 
+func IsInvalidPowerStateError(err error) bool {
+	if soap.IsVimFault(err) {
+		_, ok1 := soap.ToVimFault(err).(*types.InvalidPowerState)
+		_, ok2 := soap.ToVimFault(err).(*types.InvalidPowerStateFault)
+		return ok1 || ok2
+	}
+
+	if soap.IsSoapFault(err) {
+		_, ok1 := soap.ToSoapFault(err).VimFault().(types.InvalidPowerState)
+		_, ok2 := soap.ToSoapFault(err).VimFault().(types.InvalidPowerStateFault)
+		// sometimes we get the correct fault but wrong type
+		return ok1 || ok2 || soap.ToSoapFault(err).String == "vim.fault.InvalidArgument" ||
+			soap.ToSoapFault(err).String == "vim.fault.InvalidArgumentFault"
```

Comment:
`vim.fault.InvalidArgument` and `vim.fault.InvalidArgumentFault` don't seem like the right things to be check for in `IsInvalidPowerStateError`.

Reply:
ah thanks... the perils of copy and paste...